src/glibc_compat.rs

```diff
@@ -2,12 +2,30 @@ use core::marker::PhantomData;
 
 /// `JmpBufFields` are the accessible fields when viewed via a JmpBuf pointer.
 /// But also: You shouldn't be poking at these!
+#[cfg(any(target_arch = "x86_64", target_arch = "aarch64"))]
 #[repr(C)]
 pub struct JmpBufFields {
     _buf: [u64; 8],
     _neither_send_nor_sync: PhantomData<*const u8>,
 }
 
+/// `JmpBufFields` are the accessible fields when viewed via a JmpBuf pointer.
+/// But also: You shouldn't be poking at these!
+#[cfg(target_arch = "riscv64")]
+#[repr(C)]
+pub struct JmpBufFields {
```

Comment:
do we need to encode the fields at this level of detail here?

I.e. the other `JmpBufFields` just includes an abstract pile of `u64`'s (where the size of that type is dependent on the target OS+arch) that it expects the C runtime to manage on its own as part of the setjmp/longjmp implementation. Is there an advantage I'm not seeing to including the individual fields in the manner that you have here?

Reply:
To be honest, I was surprised to see that the glibc code was so different between different architectures (I expected to find an array of `long int`/`long long int` -- I am also wondering why they don't just use a `uint64_t`, whatever), and in the end I used the same style of glibc to keep it easier to compare.

Said that, the "similarity" with the C code was the only reason behind this. There is no issue whatsoever with alignment or padding (because all the fields perfectly align to 8 bytes), and type punning is totally fine in this case (correct me if I am wrong, but I am pretty confident in this case). Therefore I can use an array of `u64`s similarly to x86_64 and aarch64 if you want, just let me know what you prefer! :relaxed:

Reply:
yeah I agree with you that just ensuring 8-byte alignment seems like it will handle all possible issues here.

and I think it will be easier to maintain over time if we just use an array of `u64`s.

so if you can make that change, I'll r+ after that.

Reply:
Ok, now I only made the `JMP_BUF_SIZE` target specific. I added a very brief comment for one hardcoded value, it should be fine.